{
    "active_lock_reason": null,
    "assignee": null,
    "assignees": [],
    "author_association": "NONE",
    "body": "**Description of the problem**\r\nI think I may have run across a bug with fig.meca when attempting to plot from a file. Column names are being plotted in the bottom left corner, and the issue disappears when column names are removed from the file.\r\n<!-- Please be as detailed as you can when describing an issue. The more information we have, the easier it will be for us to track this down. -->\r\n\r\n**Full code that generated the error**\r\n```\r\nimport pygmt\r\nimport os\r\n\r\n# Main File Path ----------------------------------\r\nmain_dir = r'C:\\Users\\USER\\Desktop\\TESTING'\r\n\r\n# Save Path ----------------------------------\r\nsave_name = os.path.join(main_dir, 'Results', 'map_TEST.png')\r\n\r\n# Focal Mechanism Path ----------------------------------    \r\nfocal_mecha = os.path.join(main_dir, 'Data', 'selected_fm.csv')\r\n\r\n\r\nfig = pygmt.Figure()\r\n\r\nfig.basemap(\r\n    region = [-115.85, -115.35, 32.80, 33.38],\r\n    projection = 'M6i',\r\n    frame = True\r\n)\r\nfig.coast(\r\n    land = 'grey',\r\n    water = 'lightskyblue2',\r\n)\r\nfig.meca(\r\n    spec = focal_mecha,\r\n    convention = 'aki',\r\n    scale = '1.0c',\r\n    offset = True,\r\n    )\r\n\r\nfig.savefig(save_name)\r\nprint('done')\r\n```\r\n\r\n**System information**\r\nPyGMT information:\r\nversion: v0.4.0\r\nSystem information:\r\npython: 3.9.5 (tags/v3.9.5:0a7dcbd, May  3 2021, 17:27:52) [MSC v.1928 64 bit (AMD64)]\r\nexecutable: C:\\Users\\USER\\AppData\\Local\\Programs\\Python\\Python39\\python.exe\r\nmachine: Windows-10-10.0.19043-SP0\r\nDependency information:\r\nnumpy: 1.21.0\r\npandas: 1.2.5\r\nxarray: 0.18.2\r\nnetCDF4: 1.5.7\r\npackaging: 20.9\r\nghostscript: 9.54.0\r\ngmt: 6.2.0\r\nGMT library information:\r\nbinary dir: C:/Users/USER/AppData/Local/Programs/Python/Python39\r\ncores: 4\r\ngrid layout: rows\r\nlibrary path: c:/programs/gmt6/bin/gmt_w64.dll\r\npadding: 2\r\nplugin dir: c:/programs/gmt6/bin/gmt_plugins\r\nshare dir: c:/programs/gmt6/share\r\nversion: 6.2.0\r\n\r\nMy data:\r\n[selected_fm.csv](https://github.com/GenericMappingTools/pygmt/files/6758103/selected_fm.csv)\r\n\r\nMy map output:\r\n![map_TEST](https://user-images.githubusercontent.com/74040471/124348567-712d2880-db9f-11eb-8bea-84eda2afc3fa.png)\r\n\r\n",
    "closed_at": "2022-01-05T03:30:43Z",
    "closed_by": {
        "avatar_url": "https://avatars.githubusercontent.com/u/3974108?v=4",
        "events_url": "https://api.github.com/users/seisman/events{/privacy}",
        "followers_url": "https://api.github.com/users/seisman/followers",
        "following_url": "https://api.github.com/users/seisman/following{/other_user}",
        "gists_url": "https://api.github.com/users/seisman/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/seisman",
        "id": 3974108,
        "login": "seisman",
        "node_id": "MDQ6VXNlcjM5NzQxMDg=",
        "organizations_url": "https://api.github.com/users/seisman/orgs",
        "received_events_url": "https://api.github.com/users/seisman/received_events",
        "repos_url": "https://api.github.com/users/seisman/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/seisman/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/seisman/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/seisman"
    },
    "comments": 5,
    "comments_content": [
        {
            "author_association": "MEMBER",
            "body": "It's not a bug. For text files with a header like \r\n```\r\nlongitude\tlatitude\tdepth\tstrike\tdip\trake\tmagnitude\toffset_longitude\toffset_latitude\tlabel_mag\r\n```\r\nyou have to tell GMT or PyGMT to skip the header.",
            "created_at": "2021-07-03T18:57:59Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1367#issuecomment-873454520",
            "id": 873454520,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1367",
            "node_id": "MDEyOklzc3VlQ29tbWVudDg3MzQ1NDUyMA==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/873454520/reactions"
            },
            "updated_at": "2021-07-03T18:57:59Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/873454520",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/3974108?v=4",
                "events_url": "https://api.github.com/users/seisman/events{/privacy}",
                "followers_url": "https://api.github.com/users/seisman/followers",
                "following_url": "https://api.github.com/users/seisman/following{/other_user}",
                "gists_url": "https://api.github.com/users/seisman/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/seisman",
                "id": 3974108,
                "login": "seisman",
                "node_id": "MDQ6VXNlcjM5NzQxMDg=",
                "organizations_url": "https://api.github.com/users/seisman/orgs",
                "received_events_url": "https://api.github.com/users/seisman/received_events",
                "repos_url": "https://api.github.com/users/seisman/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/seisman/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/seisman/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/seisman"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "@PaulWessel I'm wondering if `meca` can be more clever about these headers.\r\n\r\nYou can run the following command to reproduce the issue. You will see the \"header line\" appear at the lower left corner of the map:\r\n```\r\ngmt meca -Sa1c -R-116/-115/32.5/33.5 -Baf -pdf map << EOF\r\nlongitude\tlatitude\tdepth\tstrike\tdip\trake\tmagnitude\toffset_longitude\toffset_latitude\tlabel_mag\r\n-115.52666 33.01917 10.11 314 85 -171 5.32 -115.52666 33.17\r\n-115.52783 33.017 8.69 143 87 158 5.44 -115.52783 32.117\r\n-115.6275 33.08567 9.29 158 86 -151 5.75 -115.6275 33.18567\r\n-115.78167 33.08183 8.2 280 86 171 6.2 -115.78167 33.18183\r\n-115.83434 33.01367 5.45 127 72 -159 6.6 -115.83434 33.11367\r\n-115.634 33.14283 5.64 190,62 -106 5.11 -115.634 33.24283\r\nEOF\r\n```\r\n\r\nBut, it seems `plot` can skip the header line quietly.\r\n```\r\ngmt plot -R-116/-115/32.5/33.5 -Baf -Sc0.2c -pdf points << EOF\r\nlongitude\tlatitude\tdepth\tstrike\tdip\trake\tmagnitude\toffset_longitude\toffset_latitude\tlabel_mag\r\n-115.52666 33.01917 10.11 314 85 -171 5.32 -115.52666 33.17\r\n-115.52783 33.017 8.69 143 87 158 5.44 -115.52783 32.117\r\n-115.6275 33.08567 9.29 158 86 -151 5.75 -115.6275 33.18567\r\n-115.78167 33.08183 8.2 280 86 171 6.2 -115.78167 33.18183\r\n-115.83434 33.01367 5.45 127 72 -159 6.6 -115.83434 33.11367\r\n-115.634 33.14283 5.64 190,62 -106 5.11 -115.634 33.24283\r\nEOF\r\n\r\n```",
            "created_at": "2021-07-03T19:03:17Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1367#issuecomment-873455274",
            "id": 873455274,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1367",
            "node_id": "MDEyOklzc3VlQ29tbWVudDg3MzQ1NTI3NA==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/873455274/reactions"
            },
            "updated_at": "2021-07-03T19:03:17Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/873455274",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/3974108?v=4",
                "events_url": "https://api.github.com/users/seisman/events{/privacy}",
                "followers_url": "https://api.github.com/users/seisman/followers",
                "following_url": "https://api.github.com/users/seisman/following{/other_user}",
                "gists_url": "https://api.github.com/users/seisman/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/seisman",
                "id": 3974108,
                "login": "seisman",
                "node_id": "MDQ6VXNlcjM5NzQxMDg=",
                "organizations_url": "https://api.github.com/users/seisman/orgs",
                "received_events_url": "https://api.github.com/users/seisman/received_events",
                "repos_url": "https://api.github.com/users/seisman/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/seisman/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/seisman/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/seisman"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "This is caused by an upstream issue and has been fixed by https://github.com/GenericMappingTools/gmt/pull/5427.",
            "created_at": "2021-07-03T22:08:13Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1367#issuecomment-873476986",
            "id": 873476986,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1367",
            "node_id": "MDEyOklzc3VlQ29tbWVudDg3MzQ3Njk4Ng==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/873476986/reactions"
            },
            "updated_at": "2021-07-03T22:08:13Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/873476986",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/3974108?v=4",
                "events_url": "https://api.github.com/users/seisman/events{/privacy}",
                "followers_url": "https://api.github.com/users/seisman/followers",
                "following_url": "https://api.github.com/users/seisman/following{/other_user}",
                "gists_url": "https://api.github.com/users/seisman/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/seisman",
                "id": 3974108,
                "login": "seisman",
                "node_id": "MDQ6VXNlcjM5NzQxMDg=",
                "organizations_url": "https://api.github.com/users/seisman/orgs",
                "received_events_url": "https://api.github.com/users/seisman/received_events",
                "repos_url": "https://api.github.com/users/seisman/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/seisman/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/seisman/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/seisman"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "So running `conda install -c conda-forge/label/dev gmt=6.3.0` or building GMT from source (see https://github.com/GenericMappingTools/gmt/blob/master/BUILDING.md) should allow users to get the patched version of GMT.\r\n\r\nNot sure if we need to add a test for this (probably not), but I think we can close this in PyGMT v0.6.0 once we bump the minimum GMT version to 6.3.0, agree?\r\n",
            "created_at": "2021-10-13T01:03:14Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1367#issuecomment-941811433",
            "id": 941811433,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1367",
            "node_id": "IC_kwDOBRZfO844Iubp",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/941811433/reactions"
            },
            "updated_at": "2021-10-13T01:03:14Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/941811433",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/23487320?v=4",
                "events_url": "https://api.github.com/users/weiji14/events{/privacy}",
                "followers_url": "https://api.github.com/users/weiji14/followers",
                "following_url": "https://api.github.com/users/weiji14/following{/other_user}",
                "gists_url": "https://api.github.com/users/weiji14/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/weiji14",
                "id": 23487320,
                "login": "weiji14",
                "node_id": "MDQ6VXNlcjIzNDg3MzIw",
                "organizations_url": "https://api.github.com/users/weiji14/orgs",
                "received_events_url": "https://api.github.com/users/weiji14/received_events",
                "repos_url": "https://api.github.com/users/weiji14/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/weiji14/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/weiji14/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/weiji14"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "I think we can close the issue without adding a test.",
            "created_at": "2022-01-04T09:04:26Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1367#issuecomment-1004633947",
            "id": 1004633947,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1367",
            "node_id": "IC_kwDOBRZfO8474X9b",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 1,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 1,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/1004633947/reactions"
            },
            "updated_at": "2022-01-04T09:04:26Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/1004633947",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/3974108?v=4",
                "events_url": "https://api.github.com/users/seisman/events{/privacy}",
                "followers_url": "https://api.github.com/users/seisman/followers",
                "following_url": "https://api.github.com/users/seisman/following{/other_user}",
                "gists_url": "https://api.github.com/users/seisman/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/seisman",
                "id": 3974108,
                "login": "seisman",
                "node_id": "MDQ6VXNlcjM5NzQxMDg=",
                "organizations_url": "https://api.github.com/users/seisman/orgs",
                "received_events_url": "https://api.github.com/users/seisman/received_events",
                "repos_url": "https://api.github.com/users/seisman/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/seisman/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/seisman/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/seisman"
            }
        }
    ],
    "comments_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1367/comments",
    "created_at": "2021-07-03T08:43:28Z",
    "events_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1367/events",
    "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1367",
    "id": 936207460,
    "labels": [
        {
            "color": "ee0701",
            "default": true,
            "description": "Something isn't working",
            "id": 563309827,
            "name": "bug",
            "node_id": "MDU6TGFiZWw1NjMzMDk4Mjc=",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/labels/bug"
        },
        {
            "color": "e8ff6b",
            "default": false,
            "description": "Bug or missing feature of upstream core GMT",
            "id": 2059525605,
            "name": "upstream",
            "node_id": "MDU6TGFiZWwyMDU5NTI1NjA1",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/labels/upstream"
        }
    ],
    "labels_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1367/labels{/name}",
    "locked": false,
    "milestone": {
        "closed_at": "2022-03-15T23:17:52Z",
        "closed_issues": 174,
        "created_at": "2021-06-23T03:22:20Z",
        "creator": {
            "avatar_url": "https://avatars.githubusercontent.com/u/3974108?v=4",
            "events_url": "https://api.github.com/users/seisman/events{/privacy}",
            "followers_url": "https://api.github.com/users/seisman/followers",
            "following_url": "https://api.github.com/users/seisman/following{/other_user}",
            "gists_url": "https://api.github.com/users/seisman/gists{/gist_id}",
            "gravatar_id": "",
            "html_url": "https://github.com/seisman",
            "id": 3974108,
            "login": "seisman",
            "node_id": "MDQ6VXNlcjM5NzQxMDg=",
            "organizations_url": "https://api.github.com/users/seisman/orgs",
            "received_events_url": "https://api.github.com/users/seisman/received_events",
            "repos_url": "https://api.github.com/users/seisman/repos",
            "site_admin": false,
            "starred_url": "https://api.github.com/users/seisman/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/seisman/subscriptions",
            "type": "User",
            "url": "https://api.github.com/users/seisman"
        },
        "description": "",
        "due_on": null,
        "html_url": "https://github.com/GenericMappingTools/pygmt/milestone/9",
        "id": 6880166,
        "labels_url": "https://api.github.com/repos/GenericMappingTools/pygmt/milestones/9/labels",
        "node_id": "MDk6TWlsZXN0b25lNjg4MDE2Ng==",
        "number": 9,
        "open_issues": 0,
        "state": "closed",
        "title": "0.6.0",
        "updated_at": "2022-03-15T23:17:52Z",
        "url": "https://api.github.com/repos/GenericMappingTools/pygmt/milestones/9"
    },
    "node_id": "MDU6SXNzdWU5MzYyMDc0NjA=",
    "number": 1367,
    "performed_via_github_app": null,
    "reactions": {
        "+1": 0,
        "-1": 0,
        "confused": 0,
        "eyes": 0,
        "heart": 0,
        "hooray": 0,
        "laugh": 0,
        "rocket": 0,
        "total_count": 0,
        "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1367/reactions"
    },
    "repository_url": "https://api.github.com/repos/GenericMappingTools/pygmt",
    "state": "closed",
    "state_reason": "completed",
    "timeline_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1367/timeline",
    "title": "Possible text bug with fig.meca when plotting from a file",
    "updated_at": "2022-01-05T03:30:43Z",
    "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1367",
    "user": {
        "avatar_url": "https://avatars.githubusercontent.com/u/74040471?v=4",
        "events_url": "https://api.github.com/users/hazardgoat/events{/privacy}",
        "followers_url": "https://api.github.com/users/hazardgoat/followers",
        "following_url": "https://api.github.com/users/hazardgoat/following{/other_user}",
        "gists_url": "https://api.github.com/users/hazardgoat/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/hazardgoat",
        "id": 74040471,
        "login": "hazardgoat",
        "node_id": "MDQ6VXNlcjc0MDQwNDcx",
        "organizations_url": "https://api.github.com/users/hazardgoat/orgs",
        "received_events_url": "https://api.github.com/users/hazardgoat/received_events",
        "repos_url": "https://api.github.com/users/hazardgoat/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/hazardgoat/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/hazardgoat/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/hazardgoat"
    }
}